lib/chef/provider/ifconfig/systemd.rb

```diff
@@ -0,0 +1,174 @@
+#
+# Copyright:: Copyright 2018, Chef Software, Inc
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require "ipaddr"
+require "iniparse"
+
+require "chef/mixin/which"
+require "chef/mixin/shell_out"
+
+class Chef
+  class Provider
+    class Ifconfig
+      class Systemd < Chef::Provider::Ifconfig
+        extend Chef::Mixin::Which
+        include Chef::Mixin::ShellOut
+        extend Chef::Mixin::ShellOut
+
+        provides :ifconfig do
+          systemd_unit_enabled?("systemd-networkd")
+        end
```

Comment:
since this ultimately does a shell_out on every evaluation it is going to slow everyone down evaluating this.  you could make it a bit better by short-circuiting the shell_out if the which fails, but it'll still be slow on systems that have systemd.

its possible this needs to be an `@class_var ||= systemd_unit_enabled?("systemd-networkd")` but that would mean that nobody could use chef to enable `systemd-networkd` and then use this.

Reply:
Yeah, one of the points of this as an experiment was to see what needs to happen to support systemd properly. We could short circuit this some by checking if we already think that systemd is in use, but really we need to set up some methods to query systemd better and use that consistently. We're only going to need to do this more frequently in the future.